src/native/http_request_utils.c

```diff
@@ -214,31 +214,39 @@ int aws_marshal_http_headers_to_dynamic_buffer(
     return AWS_OP_SUCCESS;
 }
 
+static inline enum aws_http_version s_unmarshal_http_request_to_get_version(struct aws_byte_cursor *request_blob) {
+    uint32_t version = 0;
+    if (!aws_byte_cursor_read_be32(request_blob, &version)) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+    return version;
+}
+
 static inline int s_unmarshal_http_request(struct aws_http_message *message, struct aws_byte_cursor *request_blob) {
     uint32_t field_len = 0;
+    if (aws_http_message_get_protocol_version(message) != AWS_HTTP_VERSION_2) {
+        if (!aws_byte_cursor_read_be32(request_blob, &field_len)) {
```

Comment:
add comment like "HTTP/1 puts method and path first, but those are just headers in HTTP/2"